### DISCUSSION TITLE
Unable to open multiple .nc file using xarray.open_mfdataset() call !! because of no time dimension in individual files (only lat & lon dim in files)

### DISCUSSION THREAD

**Discussion by HARSHBARDHANK94** [pydata/xarray#5616] [Q&A] [answered]
Hello all....herewith i have attached two snaps of error and file dimension
Though I am beginner, would like to know how to add time dimension in pre-process mode to open all the files at one calls.

![Error](https://user-images.githubusercontent.com/85699193/126041730-87a47e6c-81a7-477f-9d9f-553fcf4ba3ec.PNG)
![dimension](https://user-images.githubusercontent.com/85699193/126041745-2f3d5348-a793-4425-8e73-cbdbed5459fe.PNG)

**Comment by dcherian** [comment, marked as answer]
Hello @HARSHBARDHANK94 

You should try something like:

``` python
def preprocess(ds):
	# is time in ds.attrs ; if it's in the filename look at ds.encoding['source']
	ds["time"] = ...
	return ds


xr.open_mfdataset(..., concat_dim="time", preprocess=preprocess)
```

**Reply by HARSHBARDHANK94** [reply]
yaa worked ....thanks @dcherian

**Comment by HARSHBARDHANK94** [comment]
![resolved](https://user-images.githubusercontent.com/85699193/126053814-77088b21-f18a-41be-b013-85cfa128970e.PNG)
